alacritty_terminal/src/ansi.rs

```diff
@@ -905,6 +904,10 @@ where
             }};
         }
 
+        if has_ignored_intermediates || intermediates.len() > 1 {
+            unhandled!();
+        }
```

Comment:
This should either be above all in-function macros or before the macros, there's little reason to have it in between.

I'd propose taking the `let handler`, `let writer` and `if has_ignored_intermediates` and moving it after the `arg_or_default` macro.